inlanecruising_plugin/src/object_avoidance.cpp

```diff
@@ -0,0 +1,189 @@
+
+            double goal_pos = x_lead - goal_velocity * gap_time - 10.0; 
+
+            double initial_time = 0;
+            double initial_pos = 0.0; //relative initial position (first trajectory point)
+
+            double initial_accel = 0;
+            double goal_accel = 0;
+
+            // TODO: Add special case for when the object is not moving
+
+            double delta_v_max = abs(rwol_collision[0].object.velocity.twist.linear.x - max_trajectory_speed(original_tp.trajectory_points));
+            
+            double t_ref = (original_tp.trajectory_points[original_tp.trajectory_points.size() - 1].target_time.toSec() - original_tp.trajectory_points[0].target_time.toSec());
+
+            double t_ph = 4 * delta_v_max / maximum_deceleration_value;
+
+            double tp = 0;
+
+            if(t_ph > tpmin && t_ref < t_ph){
+                tp = t_ph;
+            }
+            else if(t_ph < tpmin){
+                tp = tpmin;
+            }
+            else {
+                tp = t_ref;
+            }
+
+            std::cout << "tp:  "<< tp << std::endl;
+
+            std::vector<double> values = quintic_coefficient_calculator::quintic_coefficient_calculator(initial_pos, 
+                                                                                                        goal_pos, 
+                                                                                                        current_speed_, 
+                                                                                                        goal_velocity, 
+                                                                                                        initial_accel, 
+                                                                                                        goal_accel, 
+                                                                                                        initial_time, 
+                                                                                                        tp);
+
+            std::vector<cav_msgs::TrajectoryPlanPoint> new_trajectory_points;
+
+            new_trajectory_points.push_back(original_tp.trajectory_points[0]);
+
+            std::vector<double> original_traj_downtracks = get_relative_downtracks(original_tp);
+
+
+            for(size_t i = 1; i < original_tp.trajectory_points.size(); i++ )
+            {            
+                double traj_target_time = i * tp / original_tp.trajectory_points.size();
+                std::cout << "traj_target_time: "<< traj_target_time << std::endl;
+                double dt_dist = polynomial_calc(values, traj_target_time);
+                double dv = polynomial_calc_d(values, traj_target_time);
+
+                std::cout << "new dist: "<< dt_dist << std::endl;
+                std::cout << "new velocity: "<< dv << std::endl; //TODO Consider case when resulting velocity is 0 or <0
```

Comment:
For these kinds of edge cases that we're leaving unconsidered for now, let's add checks and warn-level logs so that it's obvious if we're looking for it.

Reply:
added the warn. Also, these are being addressed in Yied